src/observations/topocentric.jl

```diff
@@ -1,151 +1,337 @@
-    obsposvelECI(x::RadecMPC{T}; eop::Union{EopIau1980, EopIau2000A} = eop_IAU2000A) where {T <: AbstractFloat}
-    obsposvelECI(x::RadarJPL{T}; eop::Union{EopIau1980, EopIau2000A} = eop_IAU2000A) where {T <: AbstractFloat}
-
-Return the observer's geocentric `[x, y, z, v_x, v_y, v_z]` "state" vector in Earth-Centered
-Inertial (ECI) reference frame. By default, the IAU200A Earth orientation model is used to
-transform from Earth-centered, Earth-fixed (ECEF) frame to ECI frame. Other Earth orientation
-models, such as the IAU1976/80 model, can be used by importing the
-`SatelliteToolboxTransformations.EopIau1980` type and passing it to the `eop` keyword
-argument in the function call.
-
-See also [`SatelliteToolboxBase.OrbitStateVector`](@ref) and [`SatelliteToolboxTransformations.sv_ecef_to_eci`](@ref).
-
-# Arguments
-
-- `observatory::ObservatoryMPC{T}`: observation site.
-- `et::T`: ephemeris time (TDB seconds since J2000.0 epoch).
-- `eop::Union{EopIau1980, EopIau2000A}`: Earth Orientation Parameters (eop).
-"""
-function obsposvelECI(observatory::ObservatoryMPC{T}, et::ET;
-        eop::Union{EopIau1980, EopIau2000A} = eop_IAU2000A) where {T <: AbstractFloat, ET<:Union{T,Taylor1{T},Taylor1{TaylorN{T}}}}
-    # Earth-Centered Earth-Fixed position position of observer
-    pos_ECEF = obs_pos_ECEF(observatory)
-
-    # UTC seconds
-    utc_secs = et - tdb_utc(et)
-    # Julian days UTC
-    jd_utc = JD_J2000 + utc_secs/daysec
-    # State vector
-    pv_ECEF = OrbitStateVector(jd_utc, pos_ECEF, zeros(3), zeros(3))
-
-    # Transform position/velocity from Earth-Centered Earth-fixed (ECEF) frame to Earth-Centered Inertial (ECI) frame
-    # ITRF: International Terrestrial Reference Frame
-    # GCRF: Geocentric Celestial Reference Frame
-    pv_ECI = sv_ecef_to_eci(pv_ECEF, Val(:ITRF), Val(:GCRF), jd_utc, eop)
-
-    # Inertial position
-    p_ECI = convert(Vector{eltype(pv_ECI.r)}, pv_ECI.r)
-    # Inertial velocity
-    v_ECI = convert(Vector{eltype(pv_ECI.v)}, pv_ECI.v)
-
-    # Concat position and velocity
-    return vcat(p_ECI, v_ECI)
-end
-
-function obsposvelECI(x::RadecMPC{T}; eop::Union{EopIau1980, EopIau2000A} = eop_IAU2000A) where {T <: AbstractFloat}
-    return obsposvelECI(x.observatory, datetime2et(x.date); eop)
-end
-function obsposvelECI(x::RadarJPL{T}; eop::Union{EopIau1980, EopIau2000A} = eop_IAU2000A) where {T <: AbstractFloat}
-    return obsposvelECI(x.rcvr, datetime2et(x.date); eop)
-end
+@doc raw"""
+    TimeOfDay
+
+Day/night at a particular timezone.
+
+# Fields
+- `light::Symbol`:
+    - for ground observatories: `:day` or `:night`,
+    - for space observatories: `:space`.
+- `start::DateTime`.
+- `stop::DateTime`.
+- `utc::Int`: hours from UTC.
+"""
+@auto_hash_equals struct TimeOfDay
+    light::Symbol
+    start::DateTime
+    stop::DateTime
+    utc::Int
+    function TimeOfDay(date::DateTime, observatory::ObservatoryMPC{T}) where {T <: AbstractFloat}

```

Comment:
I'm checking (slowly) the code, but I wonder about the need of this function (from the docstrings). My point is the following: If we know the observatory (and assuming an optical observation), then (1) the observatory is Earth-based, in which case the observation did take place at "local night"; (2) the observation took place from the space, in which case we don't care about the local night or day time.
What am I missing?

Reply:
You're right, in all space-based observations we do not care about the day/night, and all Earth-based observations occur at "local night". However, we must compare observations from different stations and times. To do so, we must know the UTC time of "dusk" and "dawn" at each night.

Reply:
Probably worth to add something to the docstrings about this point, something siimilar to what you said earlier: this is used to decide whether a number of observations correspond to the same observation batch, taken at the same time during the same night (or something similar to this).